src/components/UserImpact.js

```diff
@@ -25,34 +25,51 @@ const UserImpact = ({ userImpact, user }) => {
     pendingUserReferralCount,
   } = userImpact
   const userId = user.id
-  const showReward = confirmedImpact && !hasClaimedLatestReward
-  const referralRewardNotificationOpen = pendingUserReferralImpact > 0
+  const [showReward, setShowReward] = useState(
+    confirmedImpact && !hasClaimedLatestReward
+  )
+  const [
+    referralRewardNotificationOpen,
+    setReferralRewardNotificationOpen,
+  ] = useState(pendingUserReferralImpact > 0)
```

Comment:
@amaliwanag2 Why can't these remain calculated values (props only, not state)?

Reply:
@kmjennison as calculate props they're dependent on the mutation coming back so I figured using state would make them snappier on slower internet connections.  I could either keep them as props, or try adding an optimistic update to the mutation as alternatives

Reply:
@amaliwanag2 Yeah, either of those options sounds good! Optimistic updates are great but also an optimization at point, so don't spend too much time on it.

Generally, I think our code will be better off if we minimize state, especially state that mirrors calculated props. Otherwise, it can get hard to reason about weird bugs.

Reply:
sounds good, def agree on state and looking at this, this component is already very state FULL